manifests/init.pp

```diff
@@ -130,9 +130,20 @@
       $_package_file_name = "${puppet_agent::package_name}-${package_version}-1.osx${$::macosx_productversion_major}.dmg"
     } elsif $::operatingsystem == 'AIX' {
       $_aix_ver_number = regsubst($::platform_tag,'aix-(\d+\.\d+)-power','\1')
-      $aix_ver_number = $_aix_ver_number ? {
-        /^7\.2$/ => '7.1',
-        default  => $_aix_ver_number,
+      if $_aix_ver_number {
+        if $package_version =~ /^(\d+)\.(\d+)/ {
+          if ($1 == '1') or ($1 == '5' and $2 < '99') { # 5.99.z indicates pre-release puppet 6
+            $aix_ver_number = $_aix_ver_number ? {
+              /^7\.2$/ => '7.1',
+              default  => $_aix_ver_number,
+            }
+          }
+          else {
+            $aix_ver_number = '6.1'
+          }
+        } else {
+          $aix_ver_number = $_aix_ver_number
```

Comment:
This code should never be reached, but thought I'd allow for it just in case -- if anyone has ideas around cleaning this up, they're appreciated

Reply:
Might be cleaner to create a custom function that's named something like
```
is_puppet6?(agent_version)
```

Then you can rewrite this as:
```
$actual_aix_ver_number = regsubst($::platform_tag,'aix-(\d+\.\d+)-power','\1')
$aix_package_ver_number = is_puppet6?(agent_version) ? {
  true => '6.1'
  false => $actual_aix_ver_number ? {
    /7.2/ => 7.1
    default => $actual_aix_ver_number
  }
}
```

and probably do something similar for the other case